### ISSUE TITLE
bug in azurerm_maintenance_configuration, in_guest_user_patch_mode and install_patches properties

### ISSUE THREAD

**Issue by farwind** [hashicorp/terraform-provider-azurerm#20125]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.39.1

### Affected Resource(s)/Data Source(s)

azurerm_maintenance_configuration

### Terraform Configuration Files

```hcl
maintenance_configurations = {
    "example-mc-03" = {
      in_guest_user_patch_mode = "User"
      properties               = null
      scope                    = "InGuestPatch"
      visibility               = "Custom"
      install_patches = {
        "patches" = {
          reboot = "IfRequired"
           windows = {
            "windows1" = {
              classifications_to_include = ["Critical"]
              kb_numbers_to_include      = ["5021654", "5020614"]
              kb_numbers_to_exclude      = ["5017095"]
            }
          }
        }
      }
      windows = {
        "configuration_window_01" = {
          start_date_time      = "2032-06-21 10:00"
          expiration_date_time = "2032-06-22 22:00"
          duration             = "12:00"
          time_zone            = "Central European Standard Time"
          recur_every          = "Day"
        }
      }
    }
  }
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.39.1_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 102 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance.expandMaintenanceConfigurationInstallPatches({0x14001f38340?, 0x1097bd7e0?, 0x10af424c0?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance/maintenance_configuration_resource.go:425 +0x214
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance.resourceArmMaintenanceConfigurationCreateUpdate(0x14000313900, {0x10a47a340?, 0x14002317000})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance/maintenance_configuration_resource.go:258 +0x3f0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x10af4e820?, {0x10af4e820?, 0x14001616e10?}, 0xd?, {0x10a47a340?, 0x14002317000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:695 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400111d7a0, {0x10af4e820, 0x14001616e10}, 0x14000dfdd40, 0x14000313780, {0x10a47a340, 0x14002317000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0x86c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000037098, {0x10af4e820?, 0x14001616ba0?}, 0x140002d3cc0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xb70
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14001286960, {0x10af4e820?, 0x14001616330?}, 0x14001f0cfc0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:818 +0x3b8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10ab3fd20?, 0x14001286960}, {0x10af4e820, 0x14001616330}, 0x14001f0cf50, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000018000, {0x10af5e820, 0x140013b3860}, 0x14001bb8000, 0x1400117af60, 0x10f169b60, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1340 +0xb7c
google.golang.org/grpc.(*Server).handleStream(0x14000018000, {0x10af5e820, 0x140013b3860}, 0x14001bb8000, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1713 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.50.1/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.50.1/server.go:963 +0x290

Error: The terraform-provider-azurerm_v3.39.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

install_patches configuration to be used on vm (
source_image_reference {
    offer     = "WindowsServer"
    publisher = "MicrosoftWindowsServer"
    sku       = "2016-Datacenter"
    version   = "latest"
  }
) and resources created 

### Actual Behaviour

╷
│ Error: Plugin did not respond
│ 
│   with azurerm_dedicated_host_group.this,
│   on main.tf line 41, in resource "azurerm_dedicated_host_group" "this":
│   41: resource "azurerm_dedicated_host_group" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with azurerm_virtual_network.this,
│   on main.tf line 62, in resource "azurerm_virtual_network" "this":
│   62: resource "azurerm_virtual_network" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with azurerm_public_ip.this,
│   on main.tf line 120, in resource "azurerm_public_ip" "this":
│  120: resource "azurerm_public_ip" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.defaults.azurerm_maintenance_configuration.this["example-mc-03"],
│   on ../../main.tf line 29, in resource "azurerm_maintenance_configuration" "this":
│   29: resource "azurerm_maintenance_configuration" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.defaults.azurerm_maintenance_configuration.this["example-mc-02"],
│   on ../../main.tf line 29, in resource "azurerm_maintenance_configuration" "this":
│   29: resource "azurerm_maintenance_configuration" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.defaults.azurerm_maintenance_configuration.this["example-mc-01"],
│   on ../../main.tf line 29, in resource "azurerm_maintenance_configuration" "this":
│   29: resource "azurerm_maintenance_configuration" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

**Comment by lonegunmanb** [comment]
Hi @farwind, as a patch meant to fix this issue has been merged, would you please verify whether you would encounter this issue anymore? Thanks!

**Comment by pierluca** [comment]
A similar problem is still showing up for me.

### Terraform Version
1.3.7

### AzureRM Provider Version
3.39.1

### Affected Resource(s)/Data Source(s)
azurerm_maintenance_configuration

### Configuration

```
resource "azurerm_maintenance_configuration" "example" {
  name                = "${var.vm_name}-mc"
  resource_group_name = var.resource_group_name
  location            = var.location
  scope               = "InGuestPatch"

  window {
    start_date_time = var.maintenance_start_date_time
    duration        = var.maintenance_duration
    recur_every     = var.maintenance_recur_every
    time_zone       = "W. Europe Standard Time"
  }

  # Needed if scope is "InGuestPatch"
  install_patches {
    linux {
      classifications_to_include = [
        "Critical",
        "Security",
        "Other"
      ]
      package_names_mask_to_exclude = []
      package_names_mask_to_include = []
    }
    reboot = "IfRequired"
  }

  # Needed if scope is "InGuestPatch"
  in_guest_user_patch_mode = "User"
}
```

### Debug Output/Panic Output

```
Stack trace from the terraform-provider-azurerm_v3.44.1_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55c89b3]

goroutine 913 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance.getMaintenanceAssignmentVirtualMachine({0x74ff428?, 0xc001c33080?}, 0x0?, 0x0?, {0xc0022f2e10, 0x96})
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance/maintenance_assignment_virtual_machine_resource.go:194 +0x253
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance.resourceArmMaintenanceAssignmentVirtualMachineRead(0xc001f66f80, {0x615acc0?, 0xc002580c00?})
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance/maintenance_assignment_virtual_machine_resource.go:132 +0x18d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x74ff460?, {0x74ff460?, 0xc000613800?}, 0xd?, {0x615acc0?, 0xc002580c00?})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000dbd880, {0x74ff460, 0xc000613800}, 0xc0025eb930, {0x615acc0, 0xc002580c00})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0005b9998, {0x74ff460?, 0xc000831770?}, 0xc002893240)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:613 +0x4a5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000a634a0, {0x74ff460?, 0xc000afcf90?}, 0xc001c32e40)
github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x6879fa0?, 0xc000a634a0}, {0x74ff460, 0xc000afcf90}, 0xc0008c6d20, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00001e000, {0x750f940, 0xc0011e6ea0}, 0xc001f4cea0, 0xc0013e1650, 0xb9dd650, 0x0)
google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc00001e000, {0x750f940, 0xc0011e6ea0}, 0xc001f4cea0, 0x0)
google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.44.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by pierluca** [comment]
@lonegunmanb I'll do that in the next half-hour. Thank you !

**Comment by lonegunmanb** [comment]
@pierluca The patch is ready, but I'd like to verify it first before I submit it. No worries, just take your time, please feel free to ping me any time when you're ready.

**Comment by rcskosir** [comment]
@farwind Thank you for opening this issue. Have you been able to test the patch by @lonegunmanb #20129 to confirm if the nil panic has been corrected?

**Comment by rcskosir** [comment]
It looks like this has been resolved as of #20129, and the separate issue the mentioned in the comments was moved to its own issue. As such, I am going to mark this issue as closed.
